ports/nrf/mpconfigport.h

```diff
@@ -296,6 +307,11 @@ extern const struct _mp_obj_module_t ble_module;
 #define ROOT_POINTERS_SOFTPWM
 #endif
 
+#ifdef MICROPY_PY_TIME_USE_RTC_BASE
+// configure 24 bit time base for ticks
+#define MICROPY_PY_UTIME_TICKS_PERIOD (1 << 24)
```

Comment:
@dpgeorge : could you please verify that I got the  define right for a 24bit timebase?

Reply:
It looks correct, and it should be easy to test because 2**24 microseconds is only 16 seconds.  So if this were set to `1<<25` then ticks diff (of microseconds) will fail after 16 seconds.  I will test this myself to double check.

Reply:
Except I just realised there is no `ticks_us()`.... is it possible to make one using the result of `NRF_TIMER1->COUNTER`?  Of course that will only work for `MICROPY_PY_TIME_USE_TICKER_BASE`.

Reply:
> It looks correct, and it should be easy to test because 2**24 microseconds is only 16 seconds. So if this were set to 1<<25 then ticks diff (of microseconds) will fail after 16 seconds.

The timebase for the RTC is 1.0071msec using the HW prescaler, so the overflow occurs after 4.7 hours. 

Shouldn't be an issue anymore with the proposal below though: the overflows with 32kHz tick rate @ 24bits occurs every 512 sec and that would be handled by the irq. 

The actual msec value calculated there would then be given back to tick_msec(), and that's likely a 32bit value.

Reply:
> is it possible to make one using the result of NRF_TIMER1->COUNTER

As I understand the Nrf needs a CCRx register to read the counter value. See my comment https://github.com/micropython/micropython/pull/5470#issuecomment-587747871 on how I did this. I believe only the RTC's counter value can be accessed directly.

I didn't include this because I wanted to come through with the standard ticks_msec first ;) But I can go for it too.. Do we want to spend CCR1 for the usec timebase?

Reply:
I don't know anything about the Nrf port, so this may be a non-issue, but please keep in mind that a real-time clock is different from a CPU cycle counter clock. Real-time clocks can get synchronized to the real world and therefore stretched or stepped. ticks/ticks_ms/ticks_us shouldn't be affected by that.

Reply:
@tve fyi: RTC in this context stands for Real Time Counter, it's not a real world clock.

Reply:
> I didn't include this because I wanted to come through with the standard ticks_msec first

You're right, that's definitely a good goal, to just get `ticks_ms` working first.

Reply:
outdated with better solution, see below